apps/test/unit/javalab/ControlButtonsTest.js

```diff
@@ -21,49 +30,59 @@ describe('Java Lab Control Buttons Test', () => {
       isSubmittable: false,
       isSubmitted: false,
     };
+    stubRedux();
+    registerReducers({javalabView});
+    store = getStore();
+  });
+
+  afterEach(() => {
+    restoreRedux();
   });
 
   it('submit button for unsubmitted submittable level', () => {
-    const wrapper = shallow(
-      <ControlButtons {...defaultProps} isSubmittable isSubmitted={false} />
+    render(
+      <Provider store={store}>
```

Comment:
nit: IIRC, I think Liam used a custom render method to encapsulate the Provider setup (this feels more "natural" to me than including it every time). Do you think we should use one here? Something like...

```
function renderDefault(propOverrides = {}) {
  render(
    <Provider store={store}>
      <ControlButtons {...defaultProps} {...propOverrides} />
    </Provider>
  );
}
```

Reply:
good call, will update